.github/workflows/python-package.yml

```diff
@@ -0,0 +1,35 @@
+name: Python package
+
+on: [push]
+
+env:
+  MIN_CODE_COVERAGE_PERCENT: 93
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version-file: '.python-version'
+
+      - name: Install poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: 1.5.1
+
+      - name: Define a cache for the virtual environment based on the dependencies lock file
+        uses: actions/cache@v3
+        with:
+          path: ./.venv
+          key: venv-${{ hashFiles('poetry.lock') }}
+
+      - name: Install project dependencies
+        run: poetry install --without lint
```

Comment:
Is the plan to introduce linting in a later pull request?

Reply:
code linting is already included, handled by the yapf code formater and the ruff code linter (see https://github.com/lucsorel/py2puml/blob/feat/union-type-and-pre-commit/.pre-commit-config.yaml#L35-L45). It is handled directly by an integration with https://pre-commit.ci/ and not by github actions.

You can see these 2 checks at the bottom of this PR page, by clicking on the `Show all checks` link:
 
![image](https://github.com/lucsorel/py2puml/assets/3955251/13c4bcab-9ab0-44bb-8df0-ef76a7ed599d)

Reply:
that's why a lot of code-formatting changes are included in this PR. Which is not a good practice; I should have separated the union-type support feature and the pre-commit hooks in 2 separates PRs. But I also wanted to test the workflow of developing a feature while having these hooks. I hope you will forgive me.

Reply:
Figured out later that linting was enforced by yapf and ruff as a pre-commit hook 😄 Should have gone through the complete review process first before posting comments ...

Introducing automated testing on push is a great addition to this project, as well as enforcing code styling through ruff 👍 

I also agree on the fact that one issue should be tackled by pull request in the future. Having several issues/feature introduced in the same pull request will make the code-review process more complicated (or scary 😨) eventually making the process of merging a pull request way too slow.

Will keep reviewing the different commits later.

Reply:
> Having several issues/feature introduced in the same pull request will make the code-review process more complicated (or scary 😨) eventually making the process of merging a pull request way too slow.

I know, and I really would like to apologize for that. I am completely new to github actions and I needed to experiment.

Thank you so much for reviewing a feature you did not even ask for :pray: Should you review a single commit, it should be that one: https://github.com/lucsorel/py2puml/pull/51/commits/d37aa3226706f7103b76c13d55a4d461aa65b27e. Everything else is related with either CI or documentation.

Reply:
@grayfox88, let me know when you have read enough to approve the feature request. Don't forget to focus only on the https://github.com/lucsorel/py2puml/commit/d37aa3226706f7103b76c13d55a4d461aa65b27e commit.